clang/lib/Driver/Driver.cpp

```diff
@@ -4995,14 +4995,33 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
 
   // For an FPGA archive, we add the unbundling step above to take care of
   // the device side, but also unbundle here to extract the host side
-  for (const auto &LI : LinkerInputs) {
+  for (auto &LI : LinkerInputs) {
     Action *UnbundlerInput = nullptr;
+    auto wrapObject = [&] {
+      bool EarlyLink = false;
+      if (const Arg *A = Args.getLastArg(options::OPT_fsycl_link_EQ))
+        EarlyLink = A->getValue() == StringRef("early");
```

Comment:
Micro-optimization contest: `EarlyLink` could probably be declared & assigned in the outer scope (i.e. before line 4998 of the current reviosion). On a slightly more serious side, this might improve the lambda readability.

Reply:
Done.